packages/hash/api/src/model/knowledge/entity.model.ts

```diff
@@ -266,21 +266,33 @@ export default class {
     });
   }
 
-  /** @see {@link LinkModel.getAllOutgoing} */
-  async getAllOutgoingLinks(graphApi: GraphApi): Promise<LinkModel[]> {
-    return await LinkModel.getAllOutgoing(graphApi, {
-      sourceEntityModel: this,
-    });
-  }
-
-  /** @see {@link LinkModel.getOutgoing} */
-  async getOutgoingLink(
+  /**
+   * Get the outgoing links of an entity.
+   *
+   * @param params.linkTypeModel (optional) - the specific link type of the outgoing links
+   */
+  async getOutgoingLinks(
     graphApi: GraphApi,
-    params: { linkTypeModel: LinkTypeModel },
+    params?: { linkTypeModel?: LinkTypeModel },
   ): Promise<LinkModel[]> {
-    return await LinkModel.getOutgoing(graphApi, {
-      sourceEntityModel: this,
-      ...params,
+    const outgoingLinks = await LinkModel.getByQuery(graphApi, {
+      all: [
+        {
+          eq: [{ path: ["source", "id"] }, { literal: this.entityId }],
+        },
+        params?.linkTypeModel
+          ? {
+              eq: [
+                { path: ["type", "versionedUri"] },
+                {
+                  literal: params.linkTypeModel.schema.$id,
+                },
+              ],
+            }
+          : [],
+      ].flat(),
```

Comment:
Would `flat()` work as intended here? Seems that we're only dealing with a 1-dimensional array, so what would happen to the clauses on flattening?

Reply:
Any object (i.e. `{ ... }`) would remain in the array, any array (i.e. `[...]`) in the array would be flattened. This means that an empty array (i.e. `[]`) would be removed from the array entirely, which was the intended behaviour.

I think the integration tests should confirm that this is working as intended, as the method is able to optionally filter by the `linkTypeModel`.

Reply:
I see, thank you for clarifying! :)